### PR TITLE
MODLD-527: Inspect SRS MARC record to support Inventory workflow

### PR DESCRIPTION
**Pull request by askhat-abishev**

### CODE REVIEW

File: descriptors/ModuleDescriptor-template.json [hunk 1]
```diff
@@ -122,6 +122,18 @@
           "modulePermissions": []
         }
       ]
+    },
+    {
+      "id": "marc",
```
Comment (pkjacob) [inline descriptors/ModuleDescriptor-template.json:127]: Hi @askhat-abishev,

In Okapi, each block in the "providers" array represent an interface. So, here you have defined an interface named "marc", which is implemented by mod-linked-data module.

Per okapi documentation [https://github.com/folio-org/okapi/blob/master/doc/guide.md](https://github.com/folio-org/okapi/blob/master/doc/guide.md):

```
Note that we can have multiple modules providing the same interface. These can be deployed in Okapi simultaneously, but only one such module can be enabled for any given tenant at a given time
```
This is little strange. If there is another module which has a provider with id="marc", then linked-data module cannot be enabled along with that module per this documentation.

There is a good chance that another module, which has a "provider" with "id=marc" already exists. So, I think it is better to create this API under the existing "resource" interface.

May be some thing like `/resource/inventory-id/{id}/import-check`. I know this is not a good API URL. I can't think about anything better now :)

File: src/main/java/org/folio/linked/data/service/impl/MarcServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    return Optional.ofNullable(response.getBody())
+      .map(Record::getParsedRecord)
+      .map(parsedRecord -> (Map<?, ?>) parsedRecord.getContent())
+      .map(content -> (String) content.get("leader"))
+      .map(this::isMonograph)
+      .orElse(false);
+  }
+
+  private boolean isMonograph(String leader) {
+    return leader.charAt(6) == 'a' && (leader.charAt(7) == 'a' || leader.charAt(7) == 'm');
```
Comment (AndreiBordak) [inline src/main/java/org/folio/linked/data/service/impl/MarcServiceImpl.java:29]: minor: 
probably it's a bit more readable if reuse meaningful constants from:
RecordType.LANGUAGE_MATERIAL 
BibliographLevel.MONOGRAPHIC_COMPONENT_PART
BibliographLevel.MONOGRAPH_OR_ITEM
  
so this logic might look as following
```
private boolean isMonograph(String leader) {
    return isLanguageMaterial(leader.charAt(6)) 
      && isMonographicComponentPartOrItem(leader.charAt(7));
  }

  private boolean isLanguageMaterial(char typeOfRecord) {
    return typeOfRecord == RecordType.LANGUAGE_MATERIAL.value;
  }

  private boolean isMonographicComponentPartOrItem(char bibliographicLevel) {
    return bibliographicLevel == BibliographLevel.MONOGRAPHIC_COMPONENT_PART.value
      || bibliographicLevel == BibliographLevel.MONOGRAPH_OR_ITEM.value;
  }
```

Comment (AndreiBordak) [reply src/main/java/org/folio/linked/data/service/impl/MarcServiceImpl.java:29]: also with these changes we'll have several isMonograph() implementations, another one in marc4ld lib (org.folio.marc4ld.service.marc2ld.bib.MarcBib2LdMapperImpl), i think we can do refactoring later.

**Review by AndreiBordak — APPROVED** [review]

File: descriptors/ModuleDescriptor-template.json (hunk 1 above)
Comment (askhat-abishev) [reply descriptors/ModuleDescriptor-template.json]: done

File: src/main/java/org/folio/linked/data/service/impl/MarcServiceImpl.java (hunk 2 above)
Comment (askhat-abishev) [reply src/main/java/org/folio/linked/data/service/impl/MarcServiceImpl.java]: done

**Review by PBobylev — CHANGES_REQUESTED** [review]

File: descriptors/ModuleDescriptor-template.json [hunk 3]
```diff
@@ -69,6 +69,12 @@
           "pathPattern": "/resource/metadata/{inventoryId}/id",
           "permissionsRequired": [ "linked-data.resources.bib.id.get" ],
           "modulePermissions": []
+        },
+        {
+          "methods": [ "GET" ],
+          "pathPattern": "/import/{inventoryId}/check",
```
[the remark this answers is not in the document]
Comment (askhat-abishev) [reply descriptors/ModuleDescriptor-template.json]: thanks!
forgot to prepend with '/resource'
done

File: src/main/java/org/folio/linked/data/client/SrsClient.java [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+import org.springframework.cloud.openfeign.FeignClient;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+
+@FeignClient(name = "source-storage")
+public interface SrsClient {
+
+  @GetMapping(value = "/records/{inventoryId}/formatted?idType=INSTANCE")
+  ResponseEntity<Record> getSourceStorageRecordFormattedById(@PathVariable("inventoryId") String inventoryId);
```
[the remark this answers is not in the document]
Comment (askhat-abishev) [reply src/main/java/org/folio/linked/data/client/SrsClient.java]: done

File: src/main/java/org/folio/linked/data/controller/MarcController.java [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+import org.springframework.web.bind.annotation.RestController;
+
+@RequiredArgsConstructor
+@RestController
+public class MarcController implements MarcApi {
+
+  private final MarcService marcService;
+
+  @Override
+  public ResponseEntity<String> isSupportedByInventoryId(String inventoryId) {
```
[the remark this answers is not in the document]
Comment (askhat-abishev) [reply src/main/java/org/folio/linked/data/controller/MarcController.java]: done

**Review by PBobylev — APPROVED** [review]

**Pull request merged**